tests/testtortransport.nim

```diff
@@ -162,3 +167,125 @@ suite "Tor transport":
+    onStopFired = true
+
+  asyncSpawn onRunningHandler()
+  asyncSpawn onStopHandler()
+
+  check:
+    onRunningFired == false
+    onStopFired == false
+    transport.running == false
+
+  await transport.start(addrs)
+
+  # Give the event handler time to run
+  await sleepAsync(10.milliseconds)
+
+  check:
+    onRunningFired == true
+    onStopFired == false
+    transport.running == true
+
+  await transport.stop()
+
+  # Give the event handler time to run  
+  await sleepAsync(10.milliseconds)
+
+  check:
+    onRunningFired == true
+    onStopFired == true
+    transport.running == false
+
+suite "Transport Events":
+  asyncTest "onRunning with TCP transport":
+    let upgrader = Upgrade()
+    let transport = TcpTransport.new(upgrade = upgrader)
+    await testTransportEvents(
+      transport, @[MultiAddress.init("/ip4/127.0.0.1/tcp/0").tryGet()]
+    )
+
+  asyncTest "onRunning event can be awaited":
+    let upgrader = Upgrade()
+    let transport = TcpTransport.new(upgrade = upgrader)
+
+    let startFut =
+      transport.start(@[MultiAddress.init("/ip4/127.0.0.1/tcp/0").tryGet()])
+
+    await transport.onRunning.wait()
+
+    check:
+      transport.running == true
+
+    await startFut
+    await transport.stop()
+
+  asyncTest "onRunning with Memory transport":
+    let upgrader = Upgrade()
+    let transport = MemoryTransport.new(upgrade = upgrader)
+    await testTransportEvents(transport, @[])
+
+  asyncTest "onRunning with QUIC transport":
+    let upgrader = Upgrade()
+    let privateKey = PrivateKey.random(ECDSA, (newRng())[]).tryGet()
+    let transport = QuicTransport.new(upgrader, privateKey)
+    await testTransportEvents(
+      transport, @[MultiAddress.init("/ip4/127.0.0.1/udp/0/quic-v1").tryGet()]
+    )
+
+  asyncTest "onRunning with WebSocket transport":
+    let upgrader = Upgrade()
+    let transport = WsTransport.new(upgrader)
+    await testTransportEvents(
+      transport, @[MultiAddress.init("/ip4/127.0.0.1/tcp/0/ws").tryGet()]
+    )
+
+  asyncTest "onRunning with Tor transport":
+    # Setup Tor stub server
+    let torServer = initTAddress("127.0.0.1", 9050.Port)
+    let stub = TorServerStub.new()
+    stub.registerAddr(
+      "a2mncbqsbullu7thgm4e6zxda2xccmcgzmaq44oayhdtm6rav5vovcad.onion:80",
+      "/ip4/127.0.0.1/tcp/8080",
+    )
+    let stubStartFut = stub.start(torServer)
+
+    try:
+      # Give stub time to start
+      await sleepAsync(50.milliseconds)
+
+      let upgrader = Upgrade()
+      let transport = TorTransport.new(torServer, upgrade = upgrader)
+
+      # Use a proper TcpOnion3 address format that the stub knows about
+      await testTransportEvents(
+        transport,
+        @[
+          MultiAddress
+          .init(
+            "/ip4/127.0.0.1/tcp/8080/onion3/a2mncbqsbullu7thgm4e6zxda2xccmcgzmaq44oayhdtm6rav5vovcad:80"
+          )
+          .tryGet()
+        ],
+      )
+    finally:
+      await stubStartFut.cancelAndWait()
+      await stub.stop()
```

Comment:
- minimize try to smallest logic where try is needed 
- also please explain why is try needed here?